lib/retry/index.ts

```diff
@@ -0,0 +1,52 @@
+import { Retryer } from './Retryer';
+import { DEFAULT_OPTIONS, MethodOptions, RetryOptions } from './RetryOptions';
+import { WaitStrategy } from './WaitStrategy';
+
+export { RetryOptions };
+
+/**
+ * Retries the execution of a method for a given number of attempts.
+ * If the method fails to succeed after `attempts` retries, it fails
+ * with error `Retry failed.`
+ * @param attempts max number of attempts to retry execution
+ * @param options (optional) retry options
+ */
+export function retry(attempts: number, options?: RetryOptions): any {
+  return function (target: any, propertyKey: any, descriptor: PropertyDescriptor) {
+    const method: Function = descriptor.value;
+    const retryOptions = { ...DEFAULT_OPTIONS, ...options };
+    const waitStrategy = new WaitStrategy(retryOptions.waitPattern);
+
+    let retryCount: number = 0;
+
+    descriptor.value = function () {
+      const methodOptions: MethodOptions = {
+        instance: this,
+        args: arguments,
+        method: target[propertyKey],
```

Comment:
This property is not used. Do you need it?

Reply:
property `method` is used in another file `Retryer.ts`